src/init.jl

```diff
@@ -71,7 +71,9 @@ function device!(dev::CuDevice)
         CuContext(pctx)
         pop!(CuContext) # we don't maintain a stack...
     end
-    activate(ctx)       # replace the top of the stack
+    Threads.@threads for i in 1:Threads.nthreads()
```

Comment:
Hm, this adds quite a bit of overhead to something we want to be a low-cost operation. 

Additionally I could see a case where users want to do:

```
@threads for i in 1:nthreads()
   device!(CuDevice(i))
end
```

But that my be a future us problem. Is right now the design criteria that cuda calls from task should just work and
they all end up on the same device?

Reply:
Hmm right that's a valid concern. We should probably extend the lazy-initialization to a per-thread initialization then.

>  Is right now the design criteria that cuda calls from task should just work and
> they all end up on the same device?

Basically yes, but you're right that it should be possible to bind contexts to separate threads. That should be possible with the extended initialization, and not with the current design. I'll have another go.